tools/BuildAssets/psModules/CodeGenerationModules/AutoRestCodeGenerationModule/AutoRestCodeGenerationModule.psm1

```diff
@@ -0,0 +1,482 @@
+        [string] $SdkRootDirectory,
+
+        [Parameter(ParameterSetName="sdkOutputDir", Mandatory=$false)]
+        [string] $SdkGenerationDirectory,
+
+        [Parameter(Mandatory = $false)]
+        [string] $Namespace,
+
+        [Parameter(Mandatory = $false)]
+        [string] $ConfigFileTag,
+
+        [Parameter(Mandatory = $true)]
+        [string] $LocalRestSpecPath
+    )
+    if(-not [string]::IsNullOrWhiteSpace($SdkRootDirectory)) {
+        Start-CodeGeneration -ResourceProvider $ResourceProvider -LocalRestSpecPath $LocalRestSpecPath -SdkRootDirectory $SdkRootDirectory -Namespace $Namespace -ConfigFileTag $ConfigFileTag
+    }
+    else {
+        Start-CodeGeneration -ResourceProvider $ResourceProvider -LocalRestSpecPath $LocalRestSpecPath -SdkGenerationDirectory $SdkGenerationDirectory -Namespace $Namespace -ConfigFileTag $ConfigFileTag
+    }
+}
+
+function Get-LocalSdkRepoDirectory {
+    Param(
+        [string] $currPath
+    )
+
+    while(![string]::IsNullOrEmpty($(Split-Path $currPath -Parent)))
+    {
+        if(Test-Path -Path "$currPath\src\SDKs\_metadata")
+        {
+            return $currPath
+        }
+        
+        $currPath = $(Split-Path $currPath -Parent)
+    }
+    return $currPath
+}
+
+function Start-CodeGeneration {
+    Param(
+        [string] $ResourceProvider,
+        [string] $SpecsRepoFork = "Azure",
+        [string] $SpecsRepoName = "azure-rest-api-specs",
+        [string] $SpecsRepoBranch = "master",
+        [string] $AutoRestVersion = "latest",
+        [string] $SdkRootDirectory,
+        [string] $SdkDirectory,
+        [string] $SdkGenerationDirectory,
+        [string] $Namespace,
+        [string] $ConfigFileTag,
+        [string] $LocalRestSpecPath
+    )
+    $localSdkRepoDirectory = Get-LocalSdkRepoDirectory($(Get-InvokingScriptPath))
+
+    if(-not [string]::IsNullOrWhiteSpace($LocalRestSpecPath)) {
+        
+        if($localSdkRepoDirectory -ne $null)
+        {
+            Remove-Item "$localSdkRepoDirectory\src\SDKs\_metadata\$($ResourceProvider.Replace("/","_")).txt" -ErrorAction SilentlyContinue
+        }
+        
+        $logFile = [System.IO.Path]::GetTempFileName()+".txt";
+        Write-Warning "!!!!!WARNING!!!!!!" 
+        Write-Warning "======================================================================================================================================" 
+        Write-Warning "Generating code using a local REST spec at location $LocalRestSpecPath. PRs with code generated using this spec will not be accepted" 
+        Write-Warning "======================================================================================================================================"
+        
+        $configFile = $LocalRestSpecPath
+    }
+    else
+    {
+        if ($SpecsRepoName.EndsWith("-pr")) {
+            Write-Error "AutoRest cannot generate sdk from a spec in private repos."
```

Comment:
Please use autorest to generate your code from private spec repo. Code generation skipped.